### DISCUSSION TITLE
Any() -> All() Mutation

### DISCUSSION THREAD

**Discussion by BlythMeister** [stryker-mutator/stryker-net#1679] [Q&A] [answered]
I'm 99% sure i just don't understand :)

But i have some logic doing:
```
if(collection.Any(x => x.value == true))
{
    [stuff that effects return value]
}
```
This can be mutated to `All` from `Any` but i don't see that as a problem.
The intent there was that as long as 1 item in the collection matches, do the logic.
So if All of them match, this is essentially the same as `Any`.

The `All` to `Any` mutation makes sense, but i don't understand the inverse.

Is anyone able to enlighten me as to what i'm missing please before i go mad!

**Comment by rouke-broersma** [comment, marked as answer]
Your logic is that only one item in your collection needs to match the spec for the logic to be executed. Now imagine someone accidentally changes that Any to All. Now ALL items must match before the logic is executed. Is that a code change that is allowable? If not, it should be tested. That is what mutation testing points out.

**Reply by BlythMeister** [reply]
Right, followup.

What would be the right course of action to fix it?

I have 2 tests currently.

1 with no items 
1 with 1 item

I guess the test I need is with 2 items, but only 1 matching the criteria.

Therefore when changed to all, it would no longer go into the if statement?

**Reply by rouke-broersma** [reply]
Yes that sounds like the correct fix

**Reply by BlythMeister** [reply]
Great thanks!

Im very new to mutation testing so thank you for your quick and thorough response!
